Sources/Controllers/Map/Layers/OAGPXLayer.mm

```diff
@@ -258,17 +280,29 @@ - (void) drawLine:(QVector<OsmAnd::PointI> &)points gpx:(OAGPX *)gpx baseOrder:(
             
             outlineBuilder.buildAndAddToCollection(_linesCollection);
         }
-        
-        const auto colorARGB = OsmAnd::ColorARGB((int) gpx.color);
+
+        OsmAnd::FColorARGB colorARGB;
+        if (gpx.color != 0)
+        {
+            colorARGB = OsmAnd::ColorARGB((int) gpx.color);
+        }
+        else
+        {
+            if (!colors.isEmpty() && colorizationScheme == 0)
```

Comment:
Colorization scheme can only be one of the constant values COLORIZATION_NONE, COLORIZATION_GRADIENT or COLORIZATION_SOLID. No magic numbers, please.